### PR TITLE
I2C QA Test

### PR DESCRIPTION
**Pull request by CatFreed**
Adding a test project to Kyle's integration branch (PR #142)

This tests i2c via the lsm303dlhc sensor.

When the dust settles, this should be added to the test config file:
```
        {
            "name": "LSM303DLHC I2C Test",
            "build_source": "lsm303dlhc-i2c",
            "test_command": "lsm303dlhc-i2c",
            "expected_test_output": "LSM303DLHC I2C test completed successfully!"
        }
```

Misc: Fixing the default system.destDir value

### CODE REVIEW

**Comment by CatFreed** [comment]
[I2C Testing](https://trello.com/c/YxXg0Drs/42-i2c-testing)

File: test/integration/linux/lsm303dlhc-i2c/source/main.c [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+		exit(-1);
+	}
+	
+	if (init_sensor(file) != 0)
+	{
+		printf("Something went wrong\n");
+		close(file);
+		exit(1);
+	}
+
```
Comment (plauche) [inline test/integration/linux/lsm303dlhc-i2c/source/main.c]: This looks like it is missing the check to see if the set mode matches the one read back.

Comment (CatFreed) [inline test/integration/linux/lsm303dlhc-i2c/source/main.c]: ...yup.

**Comment by plauche** [comment]
@jacoffey3

File: test/integration/linux/lsm303dlhc-i2c/LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (plauche) [inline test/integration/linux/lsm303dlhc-i2c/LICENSE]: I don't think we need to add a new license and contrib doc to every new module.

File: test/integration/linux/lsm303dlhc-i2c/source/main.c [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    msleep(20);
+
+    /* Fetch the operating mode */
+    if ((ret = read_byte(file, CTRL_REG1_A, &mode)) != 0)
+    {
+        return ret;
+    }
+
+    /* Verify that the mode received matches the mode sent */
+    if (mode != (SPEED_100Hz + NORMAL_POWER + X_AXIS_ENABLE + Y_AXIS_ENABLE
```
Comment (davesims) [inline test/integration/linux/lsm303dlhc-i2c/source/main.c]: Minor code dupe nitpick: would it make sense to extract this value into a var and use here and line 95?

**Pull request merged**